File: Daftar Belanja/DaftarBelanja.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

public class DaftarBelanja {
    public static void main(String[] args) {
        ShoppingList shoppingList = new ShoppingList();
        String input;

        Scanner scanner = new Scanner(System.in);

        do {
            System.out.println("==============================");
            System.out.println("Aplikasi Daftar belanja");
            System.out.println("==============================");
            System.out.println("1. Tambah barang ke daftar belanja");
            System.out.println("2. Lihat daftar belanja");
            System.out.println("3. Edit daftar belanja");
            System.out.println("4. Hapus daftar belanja");
            System.out.println("5. Keluar");
            System.out.println("==============================");
            input = scanner.nextLine();

            switch (input) {
                case "1":
                    shoppingList.AddShoppingList();
                    break;
                case "2":
                    shoppingList.SeeShoppingList();
                    break;
                case "3":
                    shoppingList.EditShoppingList();
                    break;
                case "4":
                    shoppingList.DeleteShoppingList();
                    break;
                case "5":
                    System.out.println("Terima kasih!");
                    break;
                default:
                    System.out.println("Input tidak valid");
            }
        } while (!input.equals("5"));

        scanner.close();
    }
}

class ShoppingList {
    private int index = -1;
    private List<String> NamaBarang = new ArrayList<>();
    private List<Integer> JumlahBarang = new ArrayList<>();

    public void AddShoppingList() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Masukkan Nama Barang : ");
        String NamaBarang = scanner.nextLine();
        this.NamaBarang.add(NamaBarang);

        System.out.print("Masukkan Jumlah Barang : ");
        int JumlahBarang = Integer.parseInt(scanner.nextLine());
        this.JumlahBarang.add(JumlahBarang);

        index++;

        scanner.close();
    }

    public void SeeShoppingList() {
        for (int i = 0; i < NamaBarang.size(); i++) {
            System.out.println((i + 1) + ". " + NamaBarang.get(i) + " (" + JumlahBarang.get(i) + ")");
        }
    }

    public void EditShoppingList() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Masukkan ID Barang yang ingin diubah : ");
        int IDBarang = Integer.parseInt(scanner.nextLine());

        if (IDBarang > index || IDBarang < 0) {
            System.out.println("ID tidak ditemukan");
            scanner.close();
            return;
        }

        System.out.print("Masukkan Nama Barang baru : ");
        String NamaBarangBaru = scanner.nextLine();

        System.out.print("Masukkan Jumlah Barang baru : ");
        int JumlahBarangBaru = Integer.parseInt(scanner.nextLine());

        NamaBarang.set(IDBarang, NamaBarangBaru);
        JumlahBarang.set(IDBarang, JumlahBarangBaru);

        scanner.close();
    }

    public void DeleteShoppingList() {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Masukkan ID Barang yang ingin dihapus : ");
        int IDBarang = Integer.parseInt(scanner.nextLine());

        if (IDBarang > index || IDBarang < 0) {
            System.out.println("ID tidak ditemukan");
            scanner.close();
            return;
        }

        NamaBarang.remove(IDBarang);
        JumlahBarang.remove(IDBarang);

        scanner.close();
    }
}

File: Daftar Belanja/DaftarBelanja.py
import sys

class ShoppingList:
    def __init__(self):
        self.index = -1
        self.NamaBarang = []
        self.JumlahBarang = []

    def AddShoppingList(self):
        print('Masukkan Nama Barang : ', end='')
        NamaBarang = input()
        self.NamaBarang.append(NamaBarang)

        print('Masukkan Jumlah Barang : ', end='')
        JumlahBarang = int(input())
        self.JumlahBarang.append(JumlahBarang)

        self.index += 1

    def SeeShoppingList(self):
        for i in range(len(self.NamaBarang)):
            print(f'{i + 1}. {self.NamaBarang[i]} ({self.JumlahBarang[i]})')

    def EditShoppingList(self):
        print('Masukkan ID Barang yang ingin diubah : ', end='')
        IDBarang = int(input())

        if IDBarang > self.index or IDBarang < 0:
            print('ID tidak ditemukan')
            return

        print('Masukkan Nama Barang baru : ', end='')
        NamaBarangBaru = input()

        print('Masukkan Jumlah Barang baru : ', end='')
        JumlahBarangBaru = int(input())

        self.NamaBarang[IDBarang] = NamaBarangBaru
        self.JumlahBarang[IDBarang] = JumlahBarangBaru

    def DeleteShoppingList(self):
        print('Masukkan ID Barang yang ingin dihapus : ', end='')
        IDBarang = int(input())

        if IDBarang > self.index or IDBarang < 0:
            print('ID tidak ditemukan')
            return

        self.NamaBarang.pop(IDBarang)
        self.JumlahBarang.pop(IDBarang)

if __name__ == '__main__':
    shoppingList = ShoppingList()

    while True:
        print('==============================')
        print('Aplikasi Daftar belanja')
        print('==============================')
        print('1. Tambah barang ke daftar belanja')
        print('2. Lihat daftar belanja')
        print('3. Edit daftar belanja')
        print('4. Hapus daftar belanja')
        print('5. Keluar')
        print('==============================')
        input_choice = input()

        if input_choice == '1':
            shoppingList.AddShoppingList()
        elif input_choice == '2':
            shoppingList.SeeShoppingList()
        elif input_choice == '3':
            shoppingList.EditShoppingList()
        elif input_choice == '4':
            shoppingList.DeleteShoppingList()
        elif input_choice == '5':
            print('Terima kasih!')
            break
        else:
            print('Input tidak valid')
